erasurecoding/src/lib.rs

```diff
@@ -0,0 +1,122 @@
+use libc::{c_int, size_t};
+use std::slice;
+
+/// Reed-Solomon encode function for FFI
+/// Takes sharded original data as byte array and produces recovery shards
+/// Parameters:
+/// - original_shards: number of original data shards
+/// - recovery_shards: number of recovery shards to generate
+/// - original_shards: input byte array containing flattened shards
+/// - shard_size: size of each shard in bytes
+/// - recovery_shards_out: buffer to store generated and flattened recovery shards
+/// Returns 0 on success, -1 on error
+#[no_mangle]
+pub unsafe extern "C" fn reed_solomon_encode(
+    original_shards_count: size_t,
+    recovery_shards_count: size_t,
+    shard_size: size_t,
+    original_shards: *const u8,
+    recovery_shards_out: *mut u8,
+) -> c_int {
+    let original_shards: Vec<_> = slice::from_raw_parts(
+        original_shards,
+        original_shards_count as usize * shard_size as usize,
+    )
+    .chunks(shard_size as usize)
+    .collect();
+
+    match reed_solomon_simd::encode(
+        original_shards_count as usize,
+        recovery_shards_count as usize,
+        original_shards,
+    ) {
+        Ok(recovery) => {
+            let output_slice = slice::from_raw_parts_mut(
+                recovery_shards_out,
```

Comment:
Should we also check if that pointer is not null? `slice::from_raw_parts_mut` behavior is undefined if the pointer is null.